src/components/post-card/PostCard.js

```diff
@@ -8,57 +8,83 @@ import './PostCard.scss';
 //  в ней должно появиться свойство state
 //  в котором будет флажок showComments
 //  и будет кнопка, лейба которой будет либо "show comments" либо "hide comments"
+const buttonLabel = ['Show COMMENTS','Hide COMMENTS']
 
-export function PostCard(props) {
-  const { post, hasImage, comments = [], author } = props;
+export class PostCard extends Component {
+  constructor( props ) {
+    super( props );
 
-  console.log('hasImage', hasImage);
-  const { title, body } = post;
+    this.state = {
+      showComments: false,
+      selectedLabel: buttonLabel[0]
```

Comment:
Here you don't need to use extra property to define label
the logic of displaying 'show comments/ hide comments' is based on the value of `showComments` property

Reply:
It's not a mistake but the way so simplify code